applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.cpp

```diff
@@ -0,0 +1,670 @@
+ * @param rLeftHandSideMatrix: the elemental left hand side matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateSecondDerivativesLHS(
+    MatrixType& rLeftHandSideMatrix,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rLeftHandSideMatrix.size1() != 0)
+        rLeftHandSideMatrix.resize(0, 0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental right hand side vector for the second derivatives constributions
+ * @param rRightHandSideVector: the elemental right hand side vector
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateSecondDerivativesRHS(
+    VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rRightHandSideVector.size() != 0)
+        rRightHandSideVector.resize(0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental mass matrix
+ * @param rMassMatrix: the elemental mass matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateMassMatrix(MatrixType& rMassMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    if (rMassMatrix.size1() != 0)
+        rMassMatrix.resize(0, 0, false);
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental damping matrix
+ * @param rDampingMatrix: the elemental damping matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateDampingMatrix(MatrixType& rDampingMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+    if (rDampingMatrix.size1() != 0)
+        rDampingMatrix.resize(0, 0, false);
+}
+
+/**
+ * This method provides the place to perform checks on the completeness of the input
+ * and the compatibility with the problem options as well as the contitutive laws selected
+ * It is designed to be called only once (or anyway, not often) typically at the beginning
+ * of the calculations, so to verify that nothing is missing from the input
+ * or that no common error is found.
+ * @param rCurrentProcessInfo
+ * this method is: MANDATORY
+ */
+int SurfaceSmoothingElement::Check(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(this->Id() < 1) <<"SurfaceSmoothingElement found with Id 0 or negative" << std::endl;
+
+    KRATOS_ERROR_IF(this->GetGeometry().Area() <= 0) << "On SurfaceSmoothingElement -> "
+        << this->Id() <<  "; Area cannot be less than or equal to 0" << std::endl;
+
+      // Base class checks for positive Jacobian and Id > 0
+      int ierr = Element::Check(rCurrentProcessInfo);
+      if(ierr != 0) return ierr;
+
+      // Check that all required variables have been registered
+      KRATOS_CHECK_VARIABLE_KEY(DISTANCE)
+      KRATOS_CHECK_VARIABLE_KEY(DISTANCE_GRADIENT)
```

Comment:
If I'm not wrong, these are no longer needed after the new variables register mechanism. Can you confirm @philbucher ?

Reply:
I also think so, since the key is assigned at compile time

@pooyan-dadvand could you confirm this please? then we could clean a lot of other code

Reply:
I will comment these checks.

Reply:
It is not needed anymore. Can be safely removed from all places.